sys/net/gnrc/network_layer/ndp/gnrc_ndp.c

```diff
@@ -181,7 +181,7 @@ void gnrc_ndp_nbr_sol_handle(kernel_pid_t iface, gnrc_pktsnip_t *pkt,
             /* see https://tools.ietf.org/html/rfc6775#section-6.5.2 */
             eui64_t iid;
             ieee802154_get_iid(&iid, ar_opt->eui64.uint8, sizeof(eui64_t));
-            ipv6_addr_set_iid(&nbr_adv_dst, iid.uint64.u64);
+            ipv6_addr_set_iid(&nbr_adv_dst, byteorder_htonll((iid.uint64.u64)).u64);
```

Comment:
Why not use just `byteorder_ntohll(iid.uint64)`?

Or to prevent all that bitshifting in the first place (because `ipv6_addr_set_iid()` will just reshift it back to network-order):

```C
ipv6_addr_set_aiid(&nbr_adv_dst, iid.uint8)
```